src/render/graph.rs

```diff
@@ -440,25 +448,68 @@ impl Graph {
                 node.has_inputs_connected = false;
             }
 
-            drop(node); // release borrow of self.nodes
+            // Node is dropped.
+            if can_free {
+                // ask the processor to release its owned resources that are not thread safe
+                // i.e. AudioRenderQuantum for buffering in delay and compressor nodes
+                let mut processor = node
+                    .processor
+                    .take()
+                    .expect("A Node should always have a processor");
+
+                processor.release_resources();
+
+                // Send processor back to control thread so that it can freed
+                // or even recycled in a pool later
+                // Note that the event sender only exists for online contexts
+                //
+                // @note - we are abusing the event dispatch system here for
+                // prototyping, this should probably be cleaned out
+                if let Some(sender) = &scope.event_sender {
+                    // we don't want to block the thread, just do the best we can
+                    let _ = sender.try_send(EventDispatch::drop_processor(processor));
+                } else {
+                    drop(processor);
+                }
+            }
+
+            // release borrow of self.nodes
+            drop(node);
 
-            // Check if we can decommission this node (end of life)
             if can_free {
-                // Node is dropped, remove it from the node list
+                //  remove node from the node list
                 nodes.remove(index);
```

Comment:
You could say here,
`let node = nodes.remove(index).unwrap();`
and then move the added section above to here